static/js/stats/stats.js

```diff
@@ -55,6 +56,26 @@ function format_date(date, include_hour) {
     return month_str + ' ' + day + ', ' + year;
 }
 
+function format_date_for_last_update(last_update) {
+    last_update = new Date(last_update * 1000);
+    var locale_date = last_update.toLocaleDateString();
+    var locale_time = last_update.toLocaleTimeString();
+    return locale_time + " on " + locale_date;
+}
+
+function update_last_full_update(end_times) {
+    if (end_times.length > 0) {
+        last_full_update = Math.min(last_full_update, end_times[end_times.length - 1]);
+    }
+
+    var full_update = format_date_for_last_update(last_full_update);
```

Comment:
Stylistically, I would just include format_last_full_update in here, since both functions are a couple of lines, and having the additional variable names makes it not as readable.